packages/agent/src/connect-v2.ts

```diff
@@ -0,0 +1,199 @@
+    // known customer credential defines these
+    client_metadata       : {
+      client_uri                     : clientUri,
+      subject_syntax_types_supported : ['did:dht'],
+    },
+  });
+
+  // Sign the Request Object using the Client DID's signing key.
+  const requestJwt = await Oidc.signJwt({
+    did  : clientDid,
+    data : request,
+  });
+
+  if (!requestJwt) {
+    throw new Error('Unable to sign requestObject');
+  }
+
+  // Encryption nonce
+  const nonce = utils.randomBytes(24);
+
+  // Encrypt the Request Object JWT using the code challenge.
+  const requestObjectJwe = await Oidc.encryptAuthRequest({
+    jwt           : requestJwt,
+    codeChallenge : codeChallengeu8a,
+    nonce,
+  });
+
+  // Convert the encrypted Request Object to URLSearchParams for form encoding.
+  const formEncodedRequest = new URLSearchParams({ request: requestObjectJwe });
+
+  const pushedAuthorizationRequestEndpoint = Oidc.buildOidcUrl({
+    baseURL  : connectServerUrl,
+    endpoint : 'pushedAuthorizationRequest',
+  });
+
+  try {
+    const parResponse = await fetch(pushedAuthorizationRequestEndpoint, {
+      body    : formEncodedRequest,
+      method  : 'POST',
+      headers : {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+    });
+    if (!parResponse.ok) {
+      throw new Error(`${parResponse.status}: ${parResponse.statusText}`);
+    }
+
+    const parData: PushedAuthResponse = await parResponse.json();
+
+    // a deeplink to a web5 compatible wallet. if the wallet scans this link it should receive
+    // a route to its web5 connect provider flow and the params of where to fetch the auth request.
+    const walletURI = new URL('web5://connect/');
+    walletURI.searchParams.set(
+      'nonce',
+      Convert.uint8Array(nonce).toBase64Url()
+    );
+    walletURI.searchParams.set('request_uri', parData.request_uri);
+    walletURI.searchParams.set('client_did', clientDid.uri);
+    walletURI.searchParams.set('code_challenge', codeChallengeb64url);
+
+    // call user's callback so they can send the URI to the wallet as they see fit
+    onUriReady(walletURI.toString());
+
+    // subscribe to receiving a response from the wallet with default TTL
+    const tokenURL = Oidc.buildOidcUrl({
+      baseURL    : connectServerUrl,
+      endpoint   : 'token',
+      tokenParam : request.state,
+    });
+
+    /** ciphertext of {@link HybridAuthResponse} */
+    const authResponse = await pollWithTTL(() => fetch(tokenURL));
+
+    if (authResponse) {
+      const jwe = await authResponse?.text();
+
+      // get the pin from the user and use it as AAD to decrypt
+      const pin = await validatePin();
+      const jwt = await Oidc.decryptAuthResponse(clientDid, jwe, pin);
+      const verifiedAuthResponse = (await Oidc.verifyJwt({
+        jwt,
+      })) as HybridAuthResponse;
+
+      // return the grants for liran to
+      console.log(verifiedAuthResponse);
+    }
+  } catch (e) {
+    console.error(e);
```

Comment:
I don't agree that this is swallowing the error; it logs it. I could return the error if that's what you mean?

Reply:
My apologies, you are right, this is not "technically" swallowing all errors, but "effectively" doing so, because the caller would not be able to distinguish between success and failure.

This is typically a definite no-no. Happy to discuss further If you think this warrants exception, else please throw/don't bother catch.

Reply:
Oh so you're saying you want me to throw. Wasn't clear from what you were asking. I'll go through and clean this up